src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -261,7 +261,36 @@ void primitive_inst::realloc_if_needed() {
         max_output_layout_size = updated_params.output_layouts[0].count();
     }
     // intermediate memory allocation is required for primitives consisting of multiple kernels in dynamic case
-    allocate_internal_buffers();
+    {
+        const auto& ibuf_layouts = _impl->get_internal_buffer_layouts();
+        if (ibuf_layouts.empty())
+            return;
+        bool can_reuse_internal_buffer = (max_intermediates_memory_sizes.size() == ibuf_layouts.size());
+        if (can_reuse_internal_buffer) {
+            for (size_t i = 0; i < max_intermediates_memory_sizes.size(); ++i) {
+                if (ibuf_layouts[i].bytes_count() > max_intermediates_memory_sizes[i]) {
```

Comment:
I'm not sure, but maybe there are some cases where intermediates_memory can be reused partially? And then, it's better to have decision about memory reuse independently for each buffer

Reply:
What do you mean? It is deciding reuse for each ibuf [i]. Do you mean something else?

Reply:
I mean situation when only single buffer requires memory reallocation, but this will cause memory reallocation for all internal inst's buffers, not only for one

Reply:
I got your point. Will update it.

Reply:
Updated, please review again: https://github.com/openvinotoolkit/openvino/pull/15169/commits/5206fe37e88dd3dfb7f4fcf1eafd7891f8326736

Reply:
Looks good, thank you!